ApplicationLibCode/Application/Tools/Telemetry/RiaOpenTelemetryManager.cpp

```diff
@@ -0,0 +1,729 @@
+{
+    // Consider healthy if success rate > 90% and we've had recent successful sends
+    const auto now                  = std::chrono::steady_clock::now();
+    const auto timeSinceLastSuccess = now - lastSuccessfulSend;
+
+    return getSuccessRate() > 0.9 && timeSinceLastSuccess < std::chrono::minutes( 5 );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RiaOpenTelemetryManager& RiaOpenTelemetryManager::instance()
+{
+    static RiaOpenTelemetryManager instance;
+    return instance;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RiaOpenTelemetryManager::RiaOpenTelemetryManager()
+    : QObject( nullptr )
+{
+    m_healthMetrics.systemStartTime    = std::chrono::steady_clock::now();
+    m_healthMetrics.lastSuccessfulSend = m_healthMetrics.systemStartTime;
+
+    // Initialize Qt networking components
+    m_networkAccessManager = new QNetworkAccessManager( this );
+
+    // Initialize timers
+    m_processTimer = new QTimer( this );
+    connect( m_processTimer, &QTimer::timeout, this, &RiaOpenTelemetryManager::onProcessEventTimer );
+
+    m_healthTimer = new QTimer( this );
+    connect( m_healthTimer, &QTimer::timeout, this, &RiaOpenTelemetryManager::sendHealthSpan );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RiaOpenTelemetryManager::~RiaOpenTelemetryManager()
+{
+    shutdown();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RiaOpenTelemetryManager::initialize()
+{
+    std::lock_guard<std::mutex> lock( m_configMutex );
+
+    // Check if OpenTelemetry is disabled in preferences
+    auto* prefs = RiaPreferencesOpenTelemetry::current();
+    if ( prefs && prefs->loggingState() == RiaPreferencesOpenTelemetry::LoggingState::DISABLED )
+    {
+        RiaLogging::info( "OpenTelemetry is disabled in preferences" );
+        return false;
+    }
+
+    if ( m_initialized.load() )
+    {
+        return true;
+    }
+
+    if ( !initializeProvider() )
+    {
+        return false;
+    }
+
+    // Start event processing timer (100ms interval for responsive processing)
+    m_isShuttingDown = false;
+    m_processTimer->start( 100 );
+
+    // Start health monitoring timer (5 minutes interval)
+    if ( m_healthMonitoringEnabled )
+    {
+        m_healthTimer->start( 5 * 60 * 1000 ); // 5 minutes in milliseconds
```

Comment:
I think it is possible to use something more modern here:
```
using namespace std::chrono_literals;
std::chrono::milliseconds interval{5m};
m_healthTimer->start(interval);
```

Not sure if it is better?

Reply:
I prefer the original code, easier to read.